### ISSUE TITLE
Model architecture and hypter-parameters

### ISSUE THREAD

**Issue by b-biswas** [b-biswas/MADNESS#22]
I am opening this issue to start exchanging on a common platform regarding architecture and hyper-parameter optimization of the deep generative model.

**Comment by b-biswas** [comment]
The main issue in the MADNESS deblender is the poor reconstruction of bright galaxies.

While looking at the results for isolated galaxies, I observed the following:
1. while making predictions for values of flux for each pixel, the minimum value of predicted sigma in each pixel is 10^-4
2. The loss function is summing over the log prob for the real value of each pixel under its predicted distribution. By doing so as the model is trained and it learns to predict mu and sigma for each pixel.

The Problem:
Since the model is only summing over all pixels (no weights), it is trying it's best to optimize the sigma for the pixels outsize the galaxy/very low value of flux.

The solution:
1. my first idea was to increase the value of the minimum predicted sigma in each pixel from 10^-4 to 10^-3. and it helps!
2.  since the model was trying too hard to optimize the sigma for low flux values... I though of adding a regularization. I removed the weight of .01 attached to the KL divergence while training the VAE (So now it is an actually VAE!)

But in some cases, the high flux values are still not well reconstructed. 
Next, I am thinking of attaching a weight to each pixel given by `sqrt( pixel_val + 0.0001)`
so now pixels with higher flux value will have more weight

**Comment by b-biswas** [comment]
After running a number of tests here are my conclusions regarding the Debvader architecture:

- The Model is unnecessarily large. (both in terms of number of layers and number of filters per layer)
- adding weights to higher flux values helps in the reconstructions. Currently, my weight is: `sqrt(pixel_val) + 0.01`. The reconstructions for brighter galaxies are now much better.
- The sigma of the prior on the latent space while training the VAE is too large. When I increase the weight on KL divergence, the distribution from where galaxies are sampled is wider than the distribution when trained with a smaller KL divergence.
